modules/passkey/contracts/SafeWebAuthnSignerProxy.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.0;
+import {IP256Verifier} from "./interfaces/IP256Verifier.sol";
+
+/**
+ * @title SafeWebAuthnSignerProxy
+ * @dev A proxy contract that points to SafeWebAuthnSigner.
+ */
+contract SafeWebAuthnSignerProxy {
+    uint256 internal immutable X;
+    uint256 internal immutable Y;
+    IP256Verifier internal immutable VERIFIER;
+    address internal singleton;
+    constructor(address implementation, uint256 x, uint256 y, address verifier) {
+        singleton = implementation;
+        X = x;
+        Y = y;
+        VERIFIER = IP256Verifier(verifier);
+    }
+
+    /// @dev Fallback function forwards all transactions and returns all received return data.
+    // solhint-disable-next-line no-complex-fallback
+    fallback() external payable {
+        bytes memory data = abi.encodePacked(msg.data, X, Y, VERIFIER);
+
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            let _singleton := and(sload(0), 0xffffffffffffffffffffffffffffffffffffffff)
+            // 0xa619486e == keccak("masterCopy()"). The value is right padded to 32-bytes with 0s
+            if eq(calldataload(0), 0xa619486e00000000000000000000000000000000000000000000000000000000) {
+                mstore(0, _singleton)
+                return(0, 0x20)
+            }
```

Comment:
Because we have an immutable `singleton`, I don't think that we need to support a `masterCopy` method.

Reply:
https://github.com/safe-global/safe-modules/pull/370/commits/8f613e69f3309c6ae7c0b5772906f9fa1a0735e0